js/priority-menu.js

```diff
@@ -0,0 +1,121 @@
+(function() {
+	/**
+	 * Prepends an element to a container.
+	 *
+	 * @param {Element} container
+	 * @param {Element} element
+	 */
+	function prependElement(container, element) {
+		if (container.firstChild) {
+			return container.insertBefore(element, container.firstChild);
+		} else {
+			return container.appendChild(element);
+		}
+	}
+
+	/**
+	 * Shows an element by adding a hidden className.
+	 *
+	 * @param {Element} element
+	 */
+	function showElement(element) {
+		// classList.remove is not supported in IE11
+		element.className = element.className.replace('is-hidden', '');
+	}
+
+	/**
+	 * Hides an element by removing the hidden className.
+	 *
+	 * @param {Element} element
+	 */
+	function hideElement(element) {
+		// classList.add is not supported in IE11
+		if (!element.classList.contains('is-hidden')) {
+			element.className += ' is-hidden';
+		}
+	}
+
+	/**
+	 * Toggles the element visibility.
+	 *
+	 * @param {Element} element
+	 */
+	function toggleElementVisibility(element) {
+		if (element.classList.contains('is-hidden')) {
+			showElement(element);
+		} else {
+			hideElement(element);
+		}
+	}
+
+	var navContainer = document.querySelector('.main-navigation ');
+	// Adds the necessary UI to operate the menu.
+	navContainer.innerHTML +=
+		'<button class="main-menu-more is-hidden">More</button>' +
```

Comment:
How about adding this More button in PHP instead of JS? This would have the added benefit of facilitating translation of the label.

Reply:
Yup, we’re working that out now.

Reply:
My idea was to keep the php pure (avoid extra nodes) if the JS is not working for some reason. Your choice though